docs/config.md

```diff
@@ -74,23 +75,80 @@ Obviously, you would only set one line, don't copy them all in!
 The default is `meld`.
 
 For example:
 
-    diff-editor = "vimdiff"
+    ui.diff-editor = "vimdiff"
 
 When kdiff3 is set via:
 
-    diff-editor = "kdiff3"
+    ui.diff-editor = "kdiff3"
 
 further settings are passed on via the following:
 
-    merge-tools.kdiff3.program = "kdiff3"
+    # merge-tools.kdiff3.program = "kdiff3"      # Defaults to the name of the tool if not specified
     merge-tools.kdiff3.edit-args = ["--merge", "--cs", "CreateBakFiles=0"]
 
+## 3-way merge tools for conflict resolution
+
+The `ui.merge-editor` key specifies the tool used for three-way merge
+tools by `jj resolve`.  For example:
+
+    ui.merge-editor = "meld"  # Or "kdiff3" or "vimdiff"
+
+The "meld", "kdiff3", and "vimdiff" tools can be used out of the box,
+as long as they are installed.
+
+To use a different tool named `TOOL`, the arguments to pass to the tool
+MUST be specified in the `merge-tools.TOOL.merge-args` key. As an example
+of how to set this key and other tool configuration options, here is
+the out-of-the-box configuration of the three default tools. (There is
+no need to copy it to your config file verbatim, but you are welcome to
+customize it.)
+
+    # merge-tools.kdiff3.program  = "kdiff3"     # Defaults to the name of the tool if not specified
+    merge-tools.kdiff3.merge-args = ["$base", "$left", "$right", "-o", "$output", "--auto"]
+    merge-tools.meld.merge-args   = ["$left", "$base", "$right", "-o", "$output", "--auto-merge"]
+
+    merge-tools.vimdiff.merge-args = ["-f", "-d", "$output", "-M",
+                                      "$left", "$base", "$right",
+                                      "-c", "wincmd J", "-c", "set modifiable",
+                                      "-c", "set write"]
+    vimdiff.program = "vim"
+    merge-tools.vimdiff.merge-tool-edits-conflict-markers = true    # See below for an explanation
+
+`jj` replaces the following arguments with the appropriate file names:
+
+- `$output` (REQUIRED) is replaced with the name of the file that the
+merge tool will output. `jj` will read this file after the merge tool
+exits. The merge is considered to have failed if this file is empty,
+unchanged since `jj` wrote it, or if the merge tool exits with a non-zero
+exit code.  - `$left` and `$right` (REQUIRED) are replaced with the
+paths to two files containing the content of each side of the conflict.
+
+- `$base` (optional) is replaced with the path to a file containing the
+contents of the conflicted file in the last common ancestor of the two
+sides of the conflict.
+
+### Editing conflict markers with a tool
+By default, the merge tool starts with an empty output file. If the tool
+puts anything into the output file, and exits with the 0 exit code,
+`jj` assumes that the conflict is fully resolved. This is appropriate
+for most graphical merge tools.
+
+Tools like `vimdiff` are better suited to facilitate editing the conflict
+markers in `jj`'s default representation of the conflict. They set the
```

Comment:
nit: "better suited" is ambiguous here - does it mean that `vimdiff` is better than other tools at editing conflict markers, or that, when using `vimdiff`, it's better to edit conflict markers than to start with an empty file? I know it's the latter, but we should clarify that somehow. Maybe something like this:

>  Some tools (e.g. `vimdiff`) can present a multi-way diff but don't resolve conflict themselves. When
>  using such tools, `jj` can help you by populating the output file with conflict markers before starting
>  the merge tool (instead of leaving the output file empty and letting the merge tool fill it in). To do that,
>   set the `merge-tools.vimdiff.merge-tool-edits-conflict-markers = true` option.